src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -135,11 +139,16 @@ internal static bool DoSslHandshake(SafeSslHandle context, byte[] recvBuf, int r
         {
             sendBuf = null;
             sendCount = 0;
-            if ((recvBuf != null) && (recvCount > 0))
+
+            sendCount = context.OutputBio.TakeBytes(out sendBuf);
+            if (recvBuf == null && sendCount > 0)
             {
-                BioWrite(context.InputBio, recvBuf, recvOffset, recvCount);
+                return false;
             }
 
+            context.InputBio.SetBio(recvBuf, recvOffset, recvCount);
+            context.OutputBio.SetBio(null, true);
```

Comment:
done, named param for the null as well just to be clear, also modified the one in the Encrypt method to have the named param for the true/false